### PR TITLE
Add Room Air Conditioner device type definition

### PR DESCRIPTION
**Pull request by huangzh142**
Add Room Air Conditioner device type definition

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -1852,6 +1852,69 @@ limitations under the License.
...
+        <domain>HA</domain>
+        <typeName>Matter Room Air Conditioner</typeName>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x0072</deviceId>
+        <clusters lockOthers="true">
+            <include cluster="Identify" client="false" server="true" clientLocked="true" serverLocked="true">
+                <requireAttribute>IDENTIFY_TIME</requireAttribute>
+                <requireAttribute>IDENTIFY_TYPE</requireAttribute>
+                <requireCommand>Identify</requireCommand>
+                <requireCommand>IdentifyQuery</requireCommand>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1866]: This does not make sense.  This command is explicitly _disallowed_ in Matter.  Yes, I know other device types here probably get it wrong too; please file an issue on them and don't make the problem worse.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -1852,6 +1852,69 @@ limitations under the License.
...
+                <requireCommand>TriggerEffect</requireCommand>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1867]: I don't see where this device type makes TriggerEffect mandatory.  This seems like a bad copy/paste from the lighting device types which _do_ make it mandatory.

Again, if other device types shouldn't make this mandatory but do, file issues.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 3]
```diff
@@ -1852,6 +1852,69 @@ limitations under the License.
...
+            </include>
+            <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
+                <requireAttribute>DEVICE_TYPE_LIST</requireAttribute>
+                <requireAttribute>SERVER_LIST</requireAttribute>
+                <requireAttribute>CLIENT_LIST</requireAttribute>
+                <requireAttribute>PARTS_LIST</requireAttribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1873]: All those are mandatory in the cluster, so why do we need to list them here?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 4]
```diff
@@ -1852,6 +1852,69 @@ limitations under the License.
...
+            </include>
+            <include cluster="Binding" client="false" server="false" clientLocked="true" serverLocked="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1875]: Why?  This device type does not allow a optional Binding server.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 5]
```diff
@@ -1852,6 +1852,69 @@ limitations under the License.
...
+                <requireCommand>ViewGroup</requireCommand>
+                <requireCommand>ViewGroupResponse</requireCommand>
+                <requireCommand>GetGroupMembership</requireCommand>
+                <requireCommand>GetGroupMembershipResponse</requireCommand>
+                <requireCommand>RemoveGroup</requireCommand>
+                <requireCommand>RemoveGroupResponse</requireCommand>
+                <requireCommand>RemoveAllGroups</requireCommand>
+                <requireCommand>AddGroupIfIdentifying</requireCommand>
+            </include>
+            <include cluster="On/Off" client="false" server="true" clientLocked="true" serverLocked="true"></include>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1911]: Why do we not have requireCommand bits for this one?  I suspect in practice we should not have them for any of the bits above except insofar as this device type overrides the default cluster conformance values.  Which it only does for Identify and On/Off.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 6]
```diff
@@ -1852,6 +1852,69 @@ limitations under the License.
...
+            <include cluster="Thermostat" client="false" server="true" clientLocked="true" serverLocked="true"></include>
+            <include cluster="Fan Control" client="false" server="false" clientLocked="false" serverLocked="true"></include>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1913]: This says "Optional Fan Control client".  That does not match the spec for this device type.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 7]
```diff
@@ -1852,6 +1852,69 @@ limitations under the License.
...
+            <include cluster="Temperature Measurement" client="false" server="false" clientLocked="false" serverLocked="true"></include>
+            <include cluster="Relative Humidity Measurement" client="false" server="false" clientLocked="false" serverLocked="true"></include>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1915]: Again, why is this allowing the _client_ clusters here?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**